src/components/Compiler/Compiler.ts

```diff
@@ -22,9 +24,9 @@ export class Compiler extends BaseCompiler {
         this.classPath = path.resolve(outputDir);
         if (config) {
             this.transformName = config.transformName;
-            if (config.logicalTypes) {
-                this.logicalTypesMap = config.logicalTypes;
-            }
+            this.logicalTypesClass = config.logicalTypesClass;
+            this.logicalTypesMap = config.logicalTypes || {};
+            this.imports = config.imports || [];
```

Comment:
Several thoughts on this one.

1. do we still need `config.logicalTypes` aka. `logicalTypesMap`? I'm not sure, we might but just checking that is actually needed by the compiler.
2. Given that we can't think of another `imports` statement and `logicalTypesClass` should default to `logicalTypes` but we can't because it also implicitly serves as the boolean flag for `has logicalTypes` I'm not sure if this is the way we should implement this. I was thinking maybe something like:

```ts
logicalTypes: LogicalTypesConfig {
  className: string = 'logicalTypes'
  from: string
  // if we need the map we could even do
  map: {[key: string]: string } = {}
}
```

Reply:
makes sense.
I'll look into the case where field is both a logicalType and a record.
it causes problems if we do the mapping(because we also have to write out the interface for that record separately, cant just change the field-name)